### PR TITLE
The gem that allows to expose o3de to console to ROS 2

### PR DESCRIPTION
**Pull request by michalpelka**
It is great tool for testing, terrible for production / automation.

### CODE REVIEW

File: Gems/ExposeConsoleToRos/Code/Source/Clients/ExposeConsoleToRosSystemComponent.cpp [hunk 1]
```diff
@@ -29,34 +29,36 @@ namespace ExposeConsoleToRos
...
     }
 
     void ExposeConsoleToRosSystemComponent::Activate()
     {
+        auto ros2Node = ROS2::ROS2Interface::Get()->GetNode();
+        AZ_Assert(ros2Node, "ROS2 node is not available");
+        m_consolePublisher = ros2Node->create_publisher<std_msgs::msg::String>("o3de_console_out", 10);
+        m_consoleSubscription = ros2Node->create_subscription<std_msgs::msg::String>("o3de_console_in", 10,
+            [](const std_msgs::msg::String::SharedPtr msg) {
+                AZ_Printf("ExposeConsoleToRos", "Received message: %s", msg->data.c_str());
```
Comment (MateuszWasilewski) [inline Gems/ExposeConsoleToRos/Code/Source/Clients/ExposeConsoleToRosSystemComponent.cpp:42]: I suggest changing to `AZ_Info`

**Review by mateuszwasilewski — COMMENTED** [review]
Functionality looks correct, this gem should be used with caution. I suggest disabling the functionality all together, if build config is `release` for security.

> [!NOTE]
> Format the code

**Comment by michalpelka** [comment]
I've make functionality available to Debug and Profile builds.

**Review by jhanca-robotecai — APPROVED** [review]

File: Gems/ExposeConsoleToRos/gem.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    ],
+    "user_tags": [
+        "ExposeConsoleToRos"
+    ],
+    "platforms": [
+        ""
+    ],
+    "icon_path": "preview.png",
+    "requirements": "User's caution is advised",
+    "documentation_url": "You should not use this gem",
```
Comment (michalpelka) [inline Gems/ExposeConsoleToRos/gem.json:22]:
```suggestion
    "documentation_url": "https://github.com/RobotecAI/robotec-o3de-tools/blob/main/readme.md",
```

File: Gems/ExposeConsoleToRos/gem.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{
+    "gem_name": "ExposeConsoleToRos",
+    "version": "1.0.0",
+    "display_name": "ExposeConsoleToRos",
+    "license": "Apache-2.0",
+    "license_url": "https://opensource.org/licenses/Apache-2.0",
+    "origin": "",
+    "origin_url": "",
```
[the remark this answers is not in the document]
Comment (michalpelka) [reply Gems/ExposeConsoleToRos/gem.json]: You can make so a good deal of shady automation with this gem.

**Pull request merged**